pkg/pillar/cmd/zedmanager/handledomainmgr.go

```diff
@@ -67,6 +67,30 @@ func MaybeAddDomainConfig(ctx *zedmanagerContext,
 		disk.DisplayName = vrs.DisplayName
 		dc.DiskConfigList = append(dc.DiskConfigList, disk)
 	}
+	// let's fill some of the default values (arguably we may want controller
+	// to do this for us and give us complete config, but it is easier to
+	// fudge DomainConfig for now on our side)
+	if dc.IsContainer() {
+		if dc.Kernel == "" {
+			dc.Kernel = "/hostfs/boot/kernel"
+		}
+		if dc.Ramdisk == "" {
+			dc.Ramdisk = "/usr/lib/xen/boot/runx-initrd"
+		}
+		if dc.ExtraArgs == "" {
+			dc.ExtraArgs = "console=hvc0 root=9p-xen dhcp=1"
```

Comment:
Nit but shouldn't we append this to ExtraArgs? That is what the old code does.

Reply:
Two issues here: I'm actually OK adding ExtraArgs back to keep this PR refactoring only (nor change in current functionality). 

That said, I feel like the old code is not quite correct actually. I feel like for when we're filling out the gaps in config it is better to take all or nothing approach. For example, both root=9p-xen and dhcp=1 are effectively arguments for a runx-initrd which means that if Ramdisk is custom they may not be needed at all.

Not quite sure what the ideal condition here would be. Perhaps:
```
		if dc.Ramdisk == "" {
			dc.Ramdisk = "/usr/lib/xen/boot/runx-initrd"
                         dc.ExtraArgs = dc.ExtraArgs + "console=hvc0 root=9p-xen dhcp=1"                      
		}
```

Or does this turn too be too convoluted and all or nothing is much easier to understand?

Thoughts @eriknordmark ?

Reply:
I don't think that is easier to understand.

Right now I can't remember why we needed the ExtraArgs in the config. Perhaps it was during the initial manual tinkering on ARM? Do we ever set it in any manifest?

Reply:
> I don't think that is easier to understand.

Ok, lets go with the initial all-or-nothing approach then -- I also feel it is much simpler to reason about

> Right now I can't remember why we needed the ExtraArgs in the config. Perhaps it was during the initial manual tinkering on ARM? Do we ever set it in any manifest?

It is actually a very useful config knob for some use-cases when you're booting a custom kernel. In fact, I'm actually hoping to use this mechanism for alternative ways of booting containers on EVE with a different kernel, initrd and as a consequence custom extra args.  The only thing that makes it complicated for us here is that we need to decide whether we take over it completely or not for when we're running "stock" containers-in-a-VM.